storage_controller/src/http.rs

```diff
@@ -157,6 +157,29 @@ async fn handle_validate(req: Request<Body>) -> Result<Response<Body>, ApiError>
     json_response(StatusCode::OK, state.service.validate(validate_req).await?)
 }
 
+async fn handle_get_timeline_import_status(req: Request<Body>) -> Result<Response<Body>, ApiError> {
+    check_permissions(&req, Scope::GenerationsApi)?;
+
+    let tenant_shard_id: TenantShardId = parse_request_param(&req, "tenant_shard_id")?;
+    let timeline_id: TimelineId = parse_request_param(&req, "timeline_id")?;
+
```

Comment:
This API, and the corresponding PUT api, isn't precondition'ed on the generation.

We should use generations to make upcalls from stale attachments fail.

Not in this PR but please take on a todo in whatever issue we have to track TODOs for split brain safety. The RFC has a section on split-brain safety.

Reply:
Shall do: https://github.com/neondatabase/neon/issues/11884